test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,241 @@
+import 'package:http_mock_adapter/http_mock_adapter.dart';
+
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+import 'package:test/test.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+void main() {
+  test('Interceptor adds signature header to GET request', () async {
+    // Arrange
+    final options = RequestsSignatureOptions(
+      clientId: 'test_client_id',
+      clientSecret: 'test_client_secret',
+      headerName: 'X-Signature',
+      signaturePattern: '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}',
+    );
+
+    final dio = Dio();
+
+    final interceptor = RequestsSignatureInterceptor(options, dio);
+
+    dio.interceptors.add(interceptor);
+
+    // Act
+    final response = await dio.get('https://google.ca');
+
+    // Assert
+    expect(response.statusCode, 200);
+
+    expect(response.requestOptions.headers, contains('X-Signature'));
+  });
+
+  test('Interceptor auto-retries on clockskew', () async {
+    final clockskewMS = 6000; // Clock skew in milliseconds
+    final toleranceMS = 500; // Tolerance in milliseconds
+
+    // Arrange
+    final options = RequestsSignatureOptions(
+      clientId: 'test_client_id',
+      clientSecret: 'test_client_secret',
+      headerName: 'X-Signature',
+      signaturePattern: '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}',
+      disableAutoRetryOnClockSkew: false,
+      clockSkew: Duration(milliseconds: clockskewMS),
+    );
+
+    // Create a Dio instance with mock adapter
+    final dio = Dio(BaseOptions(
+      validateStatus: (status) => true,
+    ));
+
+    final dioAdapter = DioAdapter(dio: dio);
+    dio.httpClientAdapter = dioAdapter;
+
+    final serverDate = DateTime.now().toUtc().add(Duration(milliseconds: 6000));
+    final serverDateIsoString = serverDate.toIso8601String();
+
+    // Create the interceptor
+    final interceptor = RequestsSignatureInterceptor(
+      options,
+      dio,
+      getTime: (request) {
+        return DateTime.now().toUtc().millisecondsSinceEpoch ~/ 1000;
+      },
+      getDateHeader: () {
+        return serverDateIsoString;
+      },
+    );
+
+    // Add the interceptor to Dio
+    dio.interceptors.add(interceptor);
+
+    // Register the mock response
+    dioAdapter.onGet('https://google.ca/', (request) {
+      // Check if the server returns its date time header
+      final hasServerDateHeader = serverDateIsoString.isNotEmpty;
+
+      // Respond with unauthorized or ok based on clock skew and tolerance settings
+      if (options.disableAutoRetryOnClockSkew || toleranceMS < clockskewMS) {
+        request.reply(HttpStatus.unauthorized, {});
+      } else if (hasServerDateHeader) {
+        request.reply(HttpStatus.ok, {});
+      } else {
+        request.reply(HttpStatus.unauthorized, {});
+      }
+    });
+
+    // Act
+    final response = await dio.get('https://google.ca/');
+
+    // Assert
+    expect(response.statusCode, HttpStatus.unauthorized);
```

Comment:
Why do you expect unauthorized if the test name is `Interceptor auto-retries on clockskew` ?

Reply:
If you're trying to test the case where the second request fails, I think you would need to capture how many requests were made to your mocked server.

Reply:
In all cases, the test name should be more explicit on the scenario you're testing.

Reply:
You closed this comment but you changed the name of the test from
- `Interceptor auto-retries on clockskew` to
- `Interceptor auto-retry behavior based on clock skew configuration`

... which again doesn't tell us what is the combination of parameters and expected result that this test is using.